rotkehlchen/externalapis/gnosispay.py

```diff
@@ -355,6 +363,36 @@ def query_remote_for_tx_and_update_events(self, tx_timestamp: Timestamp) -> None
             self.write_txdata_to_db(transaction)
             self.maybe_update_event_with_api_data(transaction)
 
+    def update_events(self, tx_timestamps: dict[EVMTxHash, Timestamp]) -> None:
+        """Update the events for the given transactions.
+        Queries the API only for transactions missing from the db.
+        """
+        with self.database.conn.read_ctx() as cursor:
+            existing_txs = set()
+            for bindings, placeholders in get_query_chunks(data=list(tx_timestamps.keys())):
+                for row in cursor.execute(
+                    f'SELECT tx_hash, timestamp, merchant_name, merchant_city, country, mcc, '
+                    f'transaction_symbol, transaction_amount, billing_symbol, billing_amount, '
+                    f'reversal_symbol, reversal_amount, reversal_tx_hash FROM gnosispay_data '
+                    f'WHERE tx_hash IN ({placeholders})',
+                    bindings,
+                ):
+                    self.maybe_update_event_with_api_data(
+                        transaction=(tx := self._deserialize_transaction_from_db(row)),
+                    )
+                    existing_txs.add(tx.tx_hash)
+
+        missing_txs = list(set(tx_timestamps.keys()) - existing_txs)
+        missing_txs.sort()
+        last_queried_ts: Timestamp | None = None
+        for tx_hash in missing_txs:
+            tx_timestamp = tx_timestamps[tx_hash]
+            if last_queried_ts is not None and abs(tx_timestamp - last_queried_ts) <= GNOSIS_PAY_TX_TIMESTAMP_RANGE:  # noqa: E501
+                continue  # skip if already included in last query
```

Comment:
is this check really needed? per transaction it is 1 check always for None, a integer difference, abs value and then comparassion and finally an assignment. At least you can make last_queried_ts = 0 at the start